onnxruntime/core/optimizer/gelu_fusion.cc

```diff
@@ -44,6 +44,20 @@ static bool IsSupportedDataType(const Node& node) {
                 [root]--> Gelu ==>
 */
 Status GeluFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level, const logging::Logger& logger) const {
+  const auto& version_map = graph.DomainToVersionMap();
+  const auto& onnx_version = version_map.find(kOnnxDomain);
+  // Gelu is an official ONNX operator as of opset 20, so we can fuse in level 1 if it is available
+  bool gelu_fusion_flag = (onnx_version != version_map.end() && onnx_version->second >= 20);
+  const auto compatible_providers = GetCompatibleExecutionProviders();
+  auto op_domain = optimization_level_ == TransformerLevel::Level1 ? kOnnxDomain : kMSDomain;
+
+  if (contrib_flag_) {
+    op_domain = kMSDomain;
+  }
+  else if ((optimization_level_ == TransformerLevel::Level1 && !gelu_fusion_flag) || (optimization_level_ == TransformerLevel::Level2 && gelu_fusion_flag)) {
+    return Status::OK();
+  }
```

Comment:
There's a lot of implicit information required to understand this code currently.

Maybe if we rename `contrib_flag_` to `allow_contrib_op_in_level_1_` and do the following it will be simpler.
 
```suggestion
  // Gelu is an official ONNX operator as of opset 20, so we can fuse in level 1 if it is available
  const bool onnx_gelu_available = (onnx_version != version_map.end() && onnx_version->second >= 20);
  const bool fuse_in_level_1 = onnx_gelu_available || allow_contrib_op_in_level_1_;
  const auto op_domain = fuse_in_level_1 && onnx_gelu_available ? kOnnxDomain : kMSDomain;

  if ((optimization_level_ == TransformerLevel::Level1 && !fuse_in_level_1) || 
      (optimization_level_ == TransformerLevel::Level2 && fuse_in_level_1)) {
      return Status::OK();
  }

  const auto compatible_providers = GetCompatibleExecutionProviders();
```

Reply:
It makes sense. I will modify the code for better readability.